src/csync/csync_private.h

```diff
@@ -102,10 +104,18 @@ struct OCSYNC_EXPORT csync_s {
       void *checksum_userdata = nullptr;
 
   } callbacks;
-  c_strlist_t *excludes = nullptr;
-  
+
   OCC::SyncJournalDb *statedb;
 
+  c_strlist_t *excludes = nullptr; /* list of individual patterns collected from all exclude files */
+  struct TraversalExcludes {
+      void prepare(c_strlist_t *excludes);
+
+      QRegularExpression regexp_exclude;
+      c_strlist_t *list_patterns_with_slashes = nullptr;
+      /* FIXME ^^ at a later point use QRegularExpression too if those become popular */
+  } parsed_traversal_excludes;
```

Comment:
Maybe this should be a pointer to the ExcludedFiles object instead, and everything should be put in the ExcludeFiles class.  This would avoid the leak.   Curently , the ~ExcludeFiles take care about destroying the `excludes` pointer.
Also now i guess we could use QVector<QByteArray>  instead od c_strlist_t.